diff --git a/ext/source/modules/API/API/API.cpp b/ext/source/modules/API/API/API.cpp
index 04ba12460..3b85ce3ad 100644
--- a/ext/source/modules/API/API/API.cpp
+++ b/ext/source/modules/API/API/API.cpp
@@ -693,6 +693,7 @@ void API::init() {
     if (this->r_ocpp.size() == 1) {
 
         this->r_ocpp.at(0)->subscribe_is_connected([this](bool is_connected) {
+	    EVLOG_info << "Received OCPP connection status callback with is_connected: " << is_connected;
             std::scoped_lock lock(ocpp_data_mutex);
             if (is_connected) {
                 this->ocpp_connection_status = "connected";
@@ -702,6 +703,7 @@ void API::init() {
         });
 
         this->r_ocpp.at(0)->subscribe_charging_schedules([this, &var_ocpp_schedule](json schedule) {
+	    EVLOG_info << "Received OCPP charging schedule update: " << schedule;
             std::scoped_lock lock(ocpp_data_mutex);
             this->ocpp_charging_schedule = schedule;
             this->ocpp_charging_schedule_updated = true;
diff --git a/ext/source/modules/EVSE/OCPP201/OCPP201.cpp b/ext/source/modules/EVSE/OCPP201/OCPP201.cpp
index 769bf9f10..520f90c3d 100644
--- a/ext/source/modules/EVSE/OCPP201/OCPP201.cpp
+++ b/ext/source/modules/EVSE/OCPP201/OCPP201.cpp
@@ -1641,6 +1641,9 @@ void OCPP201::process_reservation_end(const int32_t evse_id, const int32_t conne
 }
 
 void OCPP201::publish_charging_schedules(const std::vector<ocpp::v2::CompositeSchedule>& composite_schedules) {
+    EVLOG_info << "About to publish composite charging schedules: ";
+    for (ocpp::v2::CompositeSchedule cs: composite_schedules)
+        EVLOG_info << cs << '\n';
     const auto everest_schedules = conversions::to_everest_charging_schedules(composite_schedules);
     this->p_ocpp_generic->publish_charging_schedules(everest_schedules);
 }
