--- /ext/dist/libexec/everest/modules/JsEvManager/index.js
+++ /ext/dist/libexec/everest/modules/JsEvManager/index.js
@@ -13,6 +13,8 @@
   mod.state = 'unplugged';
 
   mod.v2g_finished = false;
+  mod.iso_eamount = 85;
+  mod.iso_departure_time = null;
   mod.iso_stopped = false;
   mod.evse_maxcurrent = 0;
   mod.maxCurrent = 0;
@@ -361,6 +363,17 @@
     return false;
   });
 
+  // c.args[0] == DepartureTime, c.args[1] == eamount
+  registerCmd(mod, 'iso_set_departure_time', 2, (mod, c) => {
+    mod.iso_departure_time = (Number(c.args[0])) ? Number(c.args[0]) : null;
+    mod.iso_eamount = (Number(c.args[1])) ? Number(c.args[1]) : 60
+    
+    evlog.debug(`iso_departure_time is: ${mod.iso_departure_time}`);
+    evlog.debug(`iso_eamount ime is: ${mod.iso_eamount}`);
+    return true;
+	
+  });
+
   if (mod.uses_list.ev.length > 0) {
     registerCmd(mod, 'iso_start_v2g_session', 1, (mod, c) => {
       switch (c.args[0]) {
@@ -372,7 +385,8 @@
         default: return false;
       }
 
-      mod.uses_list.ev[0].call.start_charging({ EnergyTransferMode: mod.energymode });
+      args = { EnergyTransferMode: mod.energymode, EAmount: mod.iso_eamount, DepartureTime: mod.iso_departure_time };
+      mod.uses_list.ev[0].call.start_charging(args);
 
       return true;
     });
