diff --git /ext/dist/libexec/everest/modules/PyEvJosev/module.py /ext/dist/libexec/everest/modules/PyEvJosev/module.py
new file mode 100644
index 35a2f94f..985c6233 100644
--- /ext/dist/libexec/everest/modules/PyEvJosev/module.py
+++ /ext/dist/libexec/everest/modules/PyEvJosev/module.py
@@ -97,7 +97,10 @@ class PyEVJosevModule():
     # implementation handlers
 
     def _handler_start_charging(self, args) -> bool:
+        log.info('_handler_start_charging() args: %s' % str(args))
 
+        self._es.DepartureTime = args['DepartureTime']
+        self._es.EAmount = args['EAmount']
         self._es.EnergyTransferMode = args['EnergyTransferMode']
 
         self._ready_event.set()
