diff --git /ext/source/modules/EV/PyEvJosev/module.py /ext/source/modules/EV/PyEvJosev/module.py
index 43a4f023a..792330ece 100644
--- /ext/source/modules/EV/PyEvJosev/module.py
+++ /ext/source/modules/EV/PyEvJosev/module.py
@@ -99,7 +99,10 @@ class PyEVJosevModule():
     # implementation handlers
 
     def _handler_start_charging(self, args) -> bool:
+        log.info('_handler_start_charging() args: %s' % str(args))
 
+        self._es.DepartureTime = args['DepartureTime']
+        self._es.EAmount = args['EAmount']
         self._es.EnergyTransferMode = args['EnergyTransferMode']
 
         self._ready_event.set()