--- /ext/cache/cpm/everest-framework/d01be681d203f70289900147e87a8f4c43ff47f0/everest-framework/src/system_unix.cpp
+++ /ext/cache/cpm/everest-framework/d01be681d203f70289900147e87a8f4c43ff47f0/everest-framework/src/system_unix.cpp
@@ -190,7 +190,7 @@ std::string set_user_and_capabilities(const std::string& run_as_user, const std:
 SubProcess SubProcess::create(const std::string& run_as_user, const std::vector<std::string>& capabilities) {
     int pipefd[2];
 
-    if (pipe2(pipefd, O_CLOEXEC | O_DIRECT)) {
+    if (pipe2(pipefd, O_CLOEXEC)) {
         throw std::runtime_error(fmt::format("Syscall pipe2() failed ({}), exiting", strerror(errno)));
     }
