diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea72c51..24eb4f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ Licence Agreement (“Licence Agreement”), clauses 1. ISO’s Copyright, \
 option(ISO15118_INSTALL "Enable install target" ${EVC_MAIN_PROJECT})
 
 # list of compile options
-set(ISO15118_COMPILE_OPTIONS_WARNING "-Wall;-Wextra;-Wno-unused-function;-Werror" CACHE STRING "A list of compile options used")
+set(ISO15118_COMPILE_OPTIONS_WARNING "-Wall;-Wextra;-Wno-unused-function" CACHE STRING "A list of compile options used")
 message(STATUS "Building libiso15118 with the following compile options: ${ISO15118_COMPILE_OPTIONS_WARNING}")
 
 if((${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME} OR ${PROJECT_NAME}_BUILD_TESTING) AND BUILD_TESTING)
